streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -168,10 +169,66 @@ public String queryableStoreName() {
                                 builder);
     }
 
+    private KTable<K, V> doFilter(final Predicate<? super K, ? super V> predicate,
+                                  final MaterializedKVInternal<K, V> materializedInternal,
+                                  final boolean filterNot) {
+        final Serde<K> keySerde;
+        final Serde<V> valueSerde;
+        final String queryableStoreName;
+        final StoreBuilder<TimestampedKeyValueStore<K, V>> storeBuilder;
```

Comment:
We need to change the type of store builder based on given materializedKV right?

Reply:
No, that's the advantage of this approach. The filter processor needs a K/V interface, which our adapter provides. As soon as we passed it through the adapter, it now _is_ a key value store.